Integrate emergency numbers

DaveTrost opened the pull request:
Integrate emergency contacts list page and emergency contacts add/edit page with the API

**Note** this PR needs to be tested with the backend branch "emergency-contacts" (or with the development branch after [PR #39](https://github.com/codeforpdx/dwellinglybackend/pull/39) is closed)

- [x] GET /emergencycontacts and populate the emergency contacts list page
- [x] use POST /emergencycontacts to add a contact after the form submission is complete for "create new emergency contact"
- [x] use DELETE /emergencycontacts/<id> to remove a contact when edit mode is selected and the delete button (minus sign) is clicked. 
  - I added a confirmation dialog before the delete is finalized. This can easily be removed if it's not needed
- [x] add a link to edit a contact. This link brings up the edit emergency contact page (same as the add contact page - but with information pre-filled). Use PUT /emergencycontacts/<id> to update the contact information in the database

KentShikama commented on src/views/addEmergencyContact.js:
```diff
@@ -82,88 +83,132 @@ const NumberSubForm = ({ i, values, errors, handleChange }) => {
+        
+        setEditMode(true);
+        axios
+            .get(`${process.env.REACT_APP_API_URL}/emergencycontacts/${id}`, makeAuthHeaders(userContext))
```
Note the axios requests will need to look like 
```
axios.get(`/api/emergencycontacts/${id}`, ...)
```
(see https://github.com/codeforpdx/dwellingly-app/pull/177)

DaveTrost replied:
Thank you for pointing this out. I rebased on development for both my front end PR and my backend PR and made this change as recommended. Everything should now be forward-compatible and deployment-ready.

KentShikama commented on src/views/addEmergencyContact.js:
```diff
@@ -82,88 +83,132 @@ const NumberSubForm = ({ i, values, errors, handleChange }) => {
+
+    const formHandler = data => {
+        const startPost = () => axios.post(`${process.env.REACT_APP_API_URL}/emergencycontacts`, data, makeAuthHeaders(userContext));
+        const startPut = () => axios.put(`${process.env.REACT_APP_API_URL}/emergencycontacts/${data.id}`, data, makeAuthHeaders(userContext));
```
Note the axios requests will need to look like 
```
axios.post("/api/emergencycontacts", ...)
axios.put(`/api/emergencycontacts/${data.id}`, ...)
```
(see https://github.com/codeforpdx/dwellingly-app/pull/177)

DaveTrost replied:
same fix as above

KentShikama commented on src/views/emergencyContacts.js:
```diff
@@ -1,91 +1,74 @@
+        const continueDelete = window.confirm('Are you sure you want to delete the emergency contact?');
+        if(!continueDelete) return;
+        axios
+            .delete(`${process.env.REACT_APP_API_URL}/emergencycontacts/${id}`, makeAuthHeaders(userContext))
```
Similarly,

```
axios.delete(`/api/emergencycontacts/${id}`, ...)
```

DaveTrost replied:
same fix as above

KentShikama commented on src/views/emergencyContacts.js:
```diff
@@ -1,91 +1,74 @@
 
+    useMountEffect(() => {
+      axios
+          .get(`${process.env.REACT_APP_API_URL}/emergencycontacts`, makeAuthHeaders(userContext))
```
Similarly,

``` 
     axios.get("/api/emergencycontacts", ...
```

DaveTrost replied:
same fix as above

DaveTrost commented:
Rebased on development and made Kent's recommended changes. Please review

KentShikama commented:
@DaveTrost I suppose if you squash merge this it will not matter but if we want to keep the diff minimal (https://github.com/codeforpdx/dwellingly-app/pull/190/files), my commits should not be included in this PR. Try undoing the merge commit with `git reset --soft HEAD~1` followed by a `git commit`.

Or feel free to pull off https://github.com/KentShikama/dwellingly-app/tree/integrate-emergency-numbers .

DaveTrost committed:
axios requests modified to comply with environment of the deployed site

add /api/ to the path for REST endpoints
remove the process environment from the path

DaveTrost committed:
Connect emergency contacts list page to API

Get data from server with GET
Make links to edit and delete contacts
Delete contacts directly with the API, after a confirmation dialog
Link to a new edit url when editing a contact
fix a styling issue that put numbers on different lines instead of the same line

DaveTrost committed:
Connect Add/Edit Emergency Contact page to API

Use a POST or PUT request to create/update the contact
For the Edit version of the page, the contact id is read from the url parameters. A state parameter prevents the form from rendering while the GET by id request processes. pre-fills the form.
Add a loading state during the POST/PUT request because otherwise the form errors display on the cleared form. On a successful request, the page navigates back to the list view
Rename numbers to contact_numbers for agreement with API parameters

DaveTrost commented:
@KentShikama, I was having a hard time removing your commits from my PR. it was easy to make the changes you suggested on my local system. However, pushing to the remote is giving errors.

EDIT: I used `git push --force` to correctly update the remote branch (see below). I hope this works without negative side-effects!

KentShikama commented:
The force push looks good to me.

Chris-Boe reviewed (approved):
Nice work!

Chris-Boe commented:
Looks great with the backend!
